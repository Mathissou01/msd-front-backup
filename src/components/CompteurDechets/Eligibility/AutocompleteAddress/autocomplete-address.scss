@use "src/styles/02-tools/tools.include-media" as im;
@use "src/styles/01-settings/settings.colors" as colors;

.c-AutocompleteAddress {
  width: 100%;
  max-width: 100%;

  &__Label {
    color: #767c97;
    font-weight: 700;
    font-size: 16px;
  }

  &__Input {
    width: 100%;
    max-width: 100%;
    height: 48px;
    margin-bottom: 16px;
    padding: 0 8px;
    border: 1px solid;
    border-color: colors.$blue-2;
    border-radius: 4px;
  }

  &__Container {
    position: relative;
  }

  &__Selector {
    width: 100%;
    max-height: 232px;
    margin-top: 2px;
    margin-bottom: 16px;
    padding: 4px;
    overflow-x: hidden;
    overflow-y: scroll;
    background: colors.$white;
    border: 2px solid colors.$blue-4;
    border-radius: 2px;
    box-shadow: 0 0 1px rgb(17 95 251 / 5%), 0 1px 3px rgb(17 95 251 / 10%);
    cursor: pointer;

    @include im.media("<=mobile") {
      position: absolute;
      bottom: calc(100% - 20px);
      z-index: 999;
      margin-top: 20px;
    }

    &::-webkit-scrollbar {
      width: 4px;
      height: 4px;
      background-color: colors.$white;
      border-radius: 15px;
    }

    &::-webkit-scrollbar-thumb {
      background-color: colors.$blue-4;
      border-radius: 15px;
    }

    &::-webkit-scrollbar-track {
      border: 1px solid colors.$blue-4;
      border-radius: 15px;
    }
  }

  &__Option {
    position: relative;
    gap: 10px;
    width: 100%;
    padding: 16px;
    text-align: left;

    &::after {
      position: absolute;
      bottom: 0;
      display: block;
      width: calc(100% - 8px);
      height: 1px;
      content: "";
    }

    &:not(:last-child) {
      border-bottom: 1px solid colors.$blue-4;
    }

    &:hover {
      background: colors.$gray-200;
    }
  }
}
