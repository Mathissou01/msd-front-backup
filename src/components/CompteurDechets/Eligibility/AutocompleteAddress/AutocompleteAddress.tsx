import React, { Dispatch, SetStateAction } from "react";
import "./autocomplete-address.scss";
import { IError } from "../../../../pages/mon-compteur-dechets/eligibilite/index.page";

interface AutocompleteAddressProps {
  value: string;
  handleChange: (e: React.ChangeEvent<HTMLInputElement>) => void;
  debouncedValue: string;
  isVisibled: boolean;
  setIsVisibled: Dispatch<SetStateAction<boolean>>;
  handleClick: (option: string) => void;
  filteredOptions: string[];
  inputName: string;
  inputLabel?: string;
  inputPlaceholder?: string;
  handleError: (updates: Partial<IError>) => void;
}

const AutocompleteAddress: React.FC<AutocompleteAddressProps> = ({
  value,
  handleChange,
  debouncedValue,
  isVisibled,
  setIsVisibled,
  handleClick,
  filteredOptions,
  inputName,
  inputLabel = "",
  inputPlaceholder = "",
  handleError,
}) => {
  return (
    <div className="c-AutocompleteAddress">
      <div className="c-AutocompleteAddress__Container">
        <label htmlFor={inputName} className="c-AutocompleteAddress__Label">
          {inputLabel} *
        </label>
        <input
          type="text"
          className="c-AutocompleteAddress__Input"
          value={value}
          name={inputName}
          onChange={handleChange}
          onFocus={() => setIsVisibled(true)}
          placeholder={inputPlaceholder}
        />
        {debouncedValue && isVisibled && (
          <div className="c-AutocompleteAddress__Selector">
            {filteredOptions.length ? (
              filteredOptions.map((option: string) => (
                <button
                  className="c-AutocompleteAddress__Option"
                  key={option}
                  onClick={() => handleClick(option)}
                >
                  {option}
                </button>
              ))
            ) : (
              <div
                className="c-AutocompleteAddress__Option"
                onClick={() =>
                  handleError({
                    isActive: true,
                    title:
                      "Malheureusement, nous ne trouvons pas votre adresse dans notre base de donnÃ©es",
                    isAddressVisible: false,
                    isReasonVisible: false,
                    isContactVisible: true,
                  })
                }
              >
                <p>
                  <strong>{"Mon adresse n'est pas dans la liste"}</strong>
                </p>
              </div>
            )}
          </div>
        )}
      </div>
    </div>
  );
};

export default AutocompleteAddress;
