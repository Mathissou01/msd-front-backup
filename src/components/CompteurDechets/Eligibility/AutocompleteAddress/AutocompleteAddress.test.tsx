import { render } from "@testing-library/react";
import AutocompleteAddress from "./AutocompleteAddress";

const mock = {
  value: "common",
  handleChange: () => {
    console.log("");
  },
  debouncedValue: "common",
  isVisibled: true,
  setIsVisibled: () => {
    console.log("");
  },
  handleClick: () => {
    console.log("");
  },
  filteredOptions: ["common autocomplete 1", "common autocomplete 2"],
  inputName: "common autocomplete",
  inputLabel: "common autocomplete",
  inputPlaceholder: "common autocomplete",
  handleError: () => {
    console.log("");
  },
};

describe("AutocompleteAddress", () => {
  it("renders", () => {
    const { container } = render(
      <AutocompleteAddress
        value={mock.value}
        handleChange={mock.handleChange}
        debouncedValue={mock.debouncedValue}
        isVisibled={mock.isVisibled}
        setIsVisibled={mock.setIsVisibled}
        handleClick={mock.handleClick}
        filteredOptions={mock.filteredOptions}
        inputName={mock.inputName}
        inputLabel={mock.inputLabel}
        inputPlaceholder={mock.inputPlaceholder}
        handleError={mock.handleError}
      />,
    );
    expect(container).toMatchSnapshot();
  });
});
