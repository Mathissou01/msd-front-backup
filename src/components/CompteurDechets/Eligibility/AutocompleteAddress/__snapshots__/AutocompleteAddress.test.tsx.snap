// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AutocompleteAddress renders 1`] = `
<div>
  <div
    class="c-AutocompleteAddress"
  >
    <div
      class="c-AutocompleteAddress__Container"
    >
      <label
        class="c-AutocompleteAddress__Label"
        for="common autocomplete"
      >
        common autocomplete
         *
      </label>
      <input
        class="c-AutocompleteAddress__Input"
        name="common autocomplete"
        placeholder="common autocomplete"
        type="text"
        value="common"
      />
      <div
        class="c-AutocompleteAddress__Selector"
      >
        <button
          class="c-AutocompleteAddress__Option"
        >
          common autocomplete 1
        </button>
        <button
          class="c-AutocompleteAddress__Option"
        >
          common autocomplete 2
        </button>
      </div>
    </div>
  </div>
</div>
`;
